src/stories/containers/transparency-report/transparency-auditor-comments/budget-status-select.tsx

```diff
@@ -1,23 +1,32 @@
 import React, { useState, useMemo, useCallback, useEffect, useRef } from 'react';
 import styled from '@emotion/styled';
 import { useThemeContext } from '../../../../core/context/ThemeContext';
-import { ExpenseReportStatus } from '../../../../core/enums/expense-reports-status.enum';
-import ExpenseReportStatusBtn from './expense-report-status-label';
+import ExpenseReportStatus from '../common/expense-report-status/expense-report-status';
+import { BudgetStatus } from '../../../../core/models/dto/core-unit.dto';
 
 export type BudgetStatusSelectProps = {
-  availableStatuses?: ExpenseReportStatus[];
+  availableStatuses?: BudgetStatus[];
+  onChangeStatus?: (status: BudgetStatus) => void;
 };
 
-const BudgetStatusSelect: React.FC<BudgetStatusSelectProps> = ({ availableStatuses }) => {
+const BudgetStatusSelect: React.FC<BudgetStatusSelectProps> = ({ availableStatuses, onChangeStatus }) => {
   const { isLight } = useThemeContext();
   const menuRef = useRef<HTMLDivElement>(null);
   const [opened, setOpened] = useState<boolean>(false);
-  const statuses = useMemo<ExpenseReportStatus[]>(() => {
-    return availableStatuses || Object.values(ExpenseReportStatus);
+  const statuses = useMemo<BudgetStatus[]>(() => {
+    return availableStatuses || Object.values(BudgetStatus);
   }, [availableStatuses]);
 
-  const [selectedStatus, setSelectedStatus] = useState<ExpenseReportStatus>(() =>
-    statuses.length > 0 ? statuses[0] : ExpenseReportStatus.Draft
+  const [selectedStatus, setSelectedStatus] = useState<BudgetStatus>(() =>
+    statuses.length > 0 ? statuses[0] : BudgetStatus.Draft
+  );
+
+  const selectStatusHandler = useCallback(
+    (status: BudgetStatus) => {
+      setSelectedStatus(status);
+      onChangeStatus?.(status);
+    },
+    [setSelectedStatus, onChangeStatus]
   );
```

Comment:
The useCallback hook is pointless in this case since the result of the hook is called later from within an arrow function that is the one passed on as a component propprop. We should either remove the hook or modify it's argument to be passed directly as the onClick prop of the component

```suggestion
  const selectStatusHandler = (status: BudgetStatus) => {
      setSelectedStatus(status);
      onChangeStatus?.(status);
  };
```